lib/docker/remote/bearer_auth.rb

```diff
@@ -27,15 +28,15 @@ def realm
       end
 
       def service
-        @serivce ||= auth_info.params['service']
+        @service ||= auth_info.params['service']
       end
 
       def token
         @token ||= begin
           http = Net::HTTP.new(realm.host, realm.port)
           http.use_ssl = true if realm.scheme == 'https'
 
-          url_params = { service: service }
+          url_params = { service: service, scope: "repository:#{repo}:pull" }
```

Comment:
The `:scope` hash entry will get replaced by the body of the `if` statement below, is that ok? Should the `repository:<repo>:pull` scope be added to the scopes in `www-authenticate` or just be a default?

Reply:
I'm not sure the function of the body of the if statement below, but it wasn't getting replaced for me. I'm using a Harbor registry and it seemed like the Bearer token was getting requested with no scopes. I used Pixie to inspect the conversation, but I'm not sure I have all the details right, all I could say for sure is I had several processes of different kinds having this conversation (image-reflector-controller and others) and the one that was failing was here.

When I added the repository scope, it no longer received 401 on the request that uses the bearer token (in between building the app image and starting the build for the assets image, where it needs to confirm if any prior asset image exists, it tries to list the tags and fails 401.)

I wouldn't really suggest this change in particular as it's poorly thought out (this token will undoubtedly get used for other stuff besides pulling a specific repo), it was just the quickest way for me to confirm and resolve what was wrong in this one specific conversation that seems to be the only place this gem+token gets used by me, at least for now.